### PR TITLE
fix(combobox, dropdown): honor `max-items` when it matches total items

### PR DESCRIPTION
**Pull request by jcfranco**
**Related Issue:** #6230 

## Summary

Fixes `max-items` logic (thanks @juliannemarik!) to show all items when it matches the total items.

### CODE REVIEW

**Comment by jcfranco** [comment]
**Note:** this logic is used in other components, so we'll have to update for consistency:

* [`combobox`](https://github.com/Esri/calcite-design-system/blob/8860dbbcff112759f93a37dd01636a6d17a9d2b2/packages/calcite-components/src/components/combobox/combobox.tsx#L927)
* [`sort-handle`](https://github.com/Esri/calcite-design-system/blob/23ec01b8001946ff57b88626d2ee875474660cc3/packages/calcite-components/src/components/sort-handle/sort-handle.tsx#L62) (has the property, but it seems to be unused)

File: packages/calcite-components/src/components/dropdown/dropdown.tsx [hunk 1]
```diff
@@ -480,11 +480,14 @@ export class Dropdown
...
 
     this.reposition(true);
-    const maxScrollerHeight = this.getMaxScrollerHeight();
-    scrollerEl.style.maxHeight = maxScrollerHeight > 0 ? `${maxScrollerHeight}px` : "";
+    scrollerEl.style.maxBlockSize = this.getMaxHeightStyle(this.getMaxScrollerHeight());
     this.reposition(true);
   }
 
+  private getMaxHeightStyle(maxHeight: number): string {
+    return maxHeight > 0 ? `min(${maxHeight}px, 90vh)` : "";
```
Comment (driskull) [inline packages/calcite-components/src/components/dropdown/dropdown.tsx:488]: Our menu style has this:

image

Should we make this a calcite global var or something? so we can use/reference it?

Comment (driskull) [reply packages/calcite-components/src/components/dropdown/dropdown.tsx:488]: 90vh is fine if the dropdown is at the top of the page but if its centered then 45 might be better. No strong opinion here though.

Comment (jcfranco) [reply packages/calcite-components/src/components/dropdown/dropdown.tsx:488]: > Should we make this a calcite global var or something? so we can use/reference it?

Yeah, I'll add a variable of some sort. #sharingiscaring

> 90vh is fine if the dropdown is at the top of the page but if its centered then 45 might be better. No strong opinion here though.

I'll play with those options.

**Comment by jcfranco** [comment]
@driskull Does `sort-handle` need to pass [`maxItems`](https://github.com/Esri/calcite-design-system/blob/23ec01b8001946ff57b88626d2ee875474660cc3/packages/calcite-components/src/components/sort-handle/sort-handle.tsx#L62) to its internal `dropdown`?

**Comment by driskull** [comment]
> @driskull Does sort-handle need to pass [maxItems](https://github.com/Esri/calcite-design-system/blob/23ec01b8001946ff57b88626d2ee875474660cc3/packages/calcite-components/src/components/sort-handle/sort-handle.tsx#L62) to its internal dropdown?

@jcfranco no, i'm not sure why that property is there. I don' think it should be.

**Comment by jcfranco** [comment]
@driskull I'll submit a separate issue to remove `maxItems` from `sort-handle`.

**Pull request merged**